### PR TITLE
Adds NUnit test tutorial for .NET Core F#

### PR DESCRIPTION
**Pull request by rprouse**
## Summary

The [Getting Started with Testing](https://docs.microsoft.com/en-us/dotnet/core/testing/) page for .NET Core has tutorials for xUnit and MSTest. This adds documentation and a sample project for NUnit in F#. To keep the size of the PRs manageable, I have submitted C# and Visual Basic tutorials separately.

See #1785 - Document NUnit in .NET Core Guide > Unit Testing

### CODE REVIEW

File: docs/core/testing/unit-testing-fsharp-with-nunit.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+```fsharp
+module MyMath =
+    let sumOfSquares xs = raise (System.NotImplementedException("You haven't written a test yet!"))
+```
+
+Change the directory back to the *unit-testing-with-fsharp* directory. Run [`dotnet sln add .\MathService\MathService.fsproj`](../tools/dotnet-sln.md)
+to add the class library project to the solution.
+
+ ### Install the NUnit project template
```
Comment (BillWagner) [inline docs/core/testing/unit-testing-fsharp-with-nunit.md]: H2

File: docs/core/testing/unit-testing-fsharp-with-nunit.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+ The NUnit test project templates need to be installed before creating a test project. This only needs to be done once. Run [`dotnet new -i NUnit3.DotNetNew.Template`](../tools/dotnet-new.md) to install the NUnit templates.
```
Comment (BillWagner) [inline docs/core/testing/unit-testing-fsharp-with-nunit.md]: Let's clarify that this is a per machine install.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks again.

Just a couple small nits.  I've also asked @cartermp to look at the F# to make sure it's idiomatic F#.

Thanks again, and we'll merge this shortly.

File: docs/core/testing/unit-testing-fsharp-with-nunit.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+Executing `dotnet test` fails the new test. You must update the `sumOfSquares` method to handle this new test. You must filter all the even numbers out of the sequence to make this test pass. You can do that by writing a small filter function and using `Seq.filter`:
+
+```fsharp
+let private isOdd x = x % 2 <> 0
+
+let sumOfSquares xs =
+    xs
+    |> Seq.filter isOdd |> Seq.toList
```
Comment (cartermp) [inline docs/core/testing/unit-testing-fsharp-with-nunit.md]: Nit: can we move the `|> Seq.toList` onto a new line?

**Comment by rprouse** [comment]
@BillWagner updates have been made based on the code reviews and pushed for review.

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks again @rprouse I've approved the changes, and I'll :shipit: as soon as the build finishes.

It will show up on the live site on the next publish cycle.

**Review by cartermp — APPROVED** [review]
Thanks @rprouse!

**Pull request merged**